packages/test/test-end-to-end-tests/src/test/summarizeRefreshLatestSummary.spec.ts

```diff
@@ -53,5 +48,79 @@ describeCompat(
 			await createSummarizer(provider, container, undefined, summaryVersion);
 			await provider.ensureSynchronized();
 		});
+
+		itExpects(
+			"The summarizing client will immediately refresh its own summaries",
+			[
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_start" },
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_end" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_start" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_end" },
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_start" },
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_end" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_start" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_end" },
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_start" },
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_end" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_start" },
+				{ eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_end" },
+			],
+			async () => {
+				const summaryConfig: ISummaryConfiguration = {
+					...DefaultSummaryConfiguration,
+					...{
+						maxTime: 5000 * 12,
+						maxAckWaitTime: 120000,
+						maxOps: 1,
+						initialSummarizerDelayMs: 0,
+					},
+				};
+				const testContainerConfig: ITestContainerConfig = {
+					runtimeOptions: {
+						summaryOptions: {
+							summaryConfigOverrides: summaryConfig,
+						},
+					},
+					loaderProps: {
+						configProvider: mockConfigProvider({
+							"Fluid.Summarizer.immediatelyRefreshLatestSummaryAck": true,
+						}),
+					},
+				};
+				const container = await provider.makeTestContainer(testContainerConfig);
+				const container2 = await provider.loadTestContainer(testContainerConfig);
+				const dataObject = (await container.getEntryPoint()) as ITestDataObject;
+				const dataObject2 = (await container2.getEntryPoint()) as ITestDataObject;
+				await waitForContainerConnection(container);
+				await waitForContainerConnection(container2);
+
+				// The first summary needs to be generated otherwise the summarizer will only generate one summary.
```

Comment:
Why?

Reply:
I don't know, I was running the test without generating the first summary and it would not work. I'll have to dig into this to answer.

Reply:
Specifically this doesn't work because the `retry` logic based on the call to `trySummarize` requires the `summarizerLock` to be defined. Because it's `undefined` the heuristics runner doesn't run in the `afterSummaryAction`. So essentially the first summary needs to occur to enable the summary lock to be locked, and then a refresh needs to happen in order to enable the lock. It's a race condition.